mux.go

```diff
@@ -107,6 +109,12 @@ func (mx *Mux) Options(pattern string, handlers ...interface{}) {
 	mx.handle(mOPTIONS, pattern, handlers...)
 }
 
+// NotFoundHandler sets a custom handler for the case when no routes matched
+// the given URL
+func (mx *Mux) NotFoundHandler(handler http.HandlerFunc) {
```

Comment:
Maybe we could call this just `NotFound(handler http.HandlerFunc)`, as goji or httprouter do.

@pkieltyka what do you think about the API?